test/cli.test.js

```diff
@@ -88,6 +88,42 @@ describe('CLI', () => {
     expect(() => JSON.parse(data)).not.toThrow();
   });
 
+  it('should be able to read template files with a UTF-8 BOM', async () => {
+    const utf8BOM = Buffer.from([0xEF, 0xBB, 0xBF]);
```

Comment:
As the utf8BOM should be the constant from arm-template-merge, perhaps should be exported there and imported here to improve maintenance of tests if this constant changes along time

Reply:
@jesus-seijas-sp Although a constant may be nice here (for readability and potential reuse), the chances of seeing the value of UTF8BOM change over time are... zero. 😉

Reply:
@nulltoken You're right. Only to know the context, there is no posibility that in the future this can be an UTF-16/UTF-32 or UTF-7 file?

Reply:
I agree with you. However, `bin/arm-template-merge` is not a module, but a shell script (yeah, I know that the difference is barely nonexistent).

I didn't really like the idea of adding a `module.exports` just to export a constant, and creating a new module which also exports a single const is also a little bit overkill.

The UTF-8 BOM will never change (actually, the BOM is really the ['ZERO WIDTH NO-BREAK SPACE' Unicode character](http://www.fileformat.info/info/unicode/char/FEFF/index.htm), only UTF-8-encoded), but if we decide in the future to support more encodings, that const (and others like it) would surely end up in a `encodings` module, which would fulfill your proposal.

Reply:
> @nulltoken You're right. Only to know the context, there is no posibility that in the future this can be an UTF-16/UTF-32 or UTF-7 file?

@jesus-seijas-sp According to the RFC, no, but that won't stop anybody from saving a JSON file in, say, UTF-16. However, that same person will probably have issues transmitting that same file.

If the need arises, we can add always add more input encodings, but I think it'd be bad practice.